sw/device/lib/base/hardened_status.c

```diff
@@ -9,7 +9,7 @@
 
 hardened_bool_t hardened_status_ok(status_t s) {
   if (launder32(s.value) >= 0 && launder32(s.value) == kHardenedBoolTrue) {
-    HARDENED_CHECK_EQ(s.value, kHardenedBoolTrue);
+    HARDENED_CHECK_EQ((hardened_bool_t)s.value, kHardenedBoolTrue);
```

Comment:
My mental model is we need the hardening check if it's a "sensitive branch". In this case, the "status is OK" branch is sensitive because maybe the attacker skipped the check to hide an error, so it's worth an extra check. But the "status is not OK" check will always result in aborting the operation and propagating the error, so it's not sensitive. I don't have an objection to adding a `HARDENED_CHECK_NE`, but that's why I didn't include it. Looping in @alphan  -- does that reasoning sound right to you?

Reply:
I also don't think we need to protect the failure branch -- protecting branches that increase privilege should be since failures lead to shutdown anyway.

What was the reason for the cast here? I don't think we need/want it. When cross-compiling this macro expands to an inline asm block, when compiling for the host, i.e. unit tests, the expansion already includes a `uint64_t` cast.